src/compiler/transformers/esDecorators.ts

```diff
@@ -2377,10 +2359,18 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
         );
     }
 
-    function createSymbolMetadataReference() {
-        return factory.createElementAccessExpression(
-            factory.createSuper(),
+    function createSymbolMetadataReference(classSuper: Identifier) {
+        const classSuperSymbolMetadata = factory.createElementAccessExpression(
+            classSuper,
             factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), "metadata"),
         );
+        return factory.createConditionalExpression(
+            factory.createLogicalOr(
+                factory.createTypeCheck(classSuperSymbolMetadata, "object"),
+                factory.createTypeCheck(classSuperSymbolMetadata, "function"),
+            ),
+            factory.createToken(SyntaxKind.QuestionToken), classSuperSymbolMetadata,
+            factory.createToken(SyntaxKind.ColonToken), factory.createNull(),
+        );
     }

```

Comment:
```suggestion
        return factory.createBinaryExpression(
            factory.createElementAccessExpression(
                classSuper,
                factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), "metadata"),
            ),
            SyntaxKind.QuestionQuestionToken,
            factory.createNull()
        );
    }

```

Reply:
@rbuckton Ok. Hm. It seems that introducing a _nullish coalescing expression_ breaks the reference to class `this`... I'll check it.

```ts
let D = (() => {
    var _a;
    let _classSuper = C;
    let _instanceExtraInitializers_1 = [];
    let _m_decorators;
    return _a = class D extends _classSuper {
            m() { }
            constructor() {
                super(...arguments);
                __runInitializers(this, _instanceExtraInitializers_1);
            }
        },
        (() => {
            var _a;
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create((_a = _classSuper[Symbol.metadata]) !== null && _a !== void 0 ? _a : null) : void 0;
            _m_decorators = [meta('b', 'z')];
            __esDecorate(_a, null, _m_decorators, { kind: "method", name: "m", static: false, private: false, access: { has: obj => "m" in obj, get: obj => obj.m }, metadata: _metadata }, null, _instanceExtraInitializers_1);
            if (_metadata) Object.defineProperty(_a, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        })(),
        _a;
})();
```

Reply:
Should we explicitly change the class name reference? Or is the `nullish coalescing` operator not handling upper scope names correctly in this case?

Reply:
Hmm. Oddly, it works perfectly fine in the playground:

https://www.staging-typescript.org/play?target=1&lib=lib.dom.d.ts%2Clib.esnext.d.ts&ts=5.2.0-pr-54657-9#code/PTAEAEBsEsCMC5QBMD2BbANAUwM4DssAPAFwFgAoACgGUBPNWFSUAQx1b1oEoA6NLYiyQtBoAPxiAvKDoMmlLgG4KFaHmJYATgDMWAYyygAYgFc8e4tBR5QAbwqhHoANqzGkPgKEiWAXUQAIlh6KJoioQCyXsKiAD6geCaQkMrkAL4q5CF4OMSgKCbEAA6FiLmaagDmzr6g0jWp2mYWVjb8gpQA1li0ZcQVeJUYoABuLJAmWH0DlVx2Dk6aAiaaNpQA+ogsnMPZGiSBwaHhmgDC1vvEc5IAfPPkTo+glHtExJ6CMSys7Nvczt1aLVpGMJlhUo80qkMuQKOB2ixKAByFhI4ZIwhIrgUPSQNjsU73R7wrzI2Bo0BI2hYhaONAKOwwmEFYqFHglHAAC0olFOrno7g+3kEtTYHG4PBYSgoLJK7w53N5-LkHgRX1Fv04vFg0qyeJw7ACoDeWDwSAJRKcJI6SPJ6IAXjSHk56XNbEyZYU5eyTFyeQFlYK1T4NeLeFLUrK2Qr-YGmEL1T8wzwdakcdZcsbCEVghokHUXBQAESEIsYYu0MvF0vl8hF+1V8i+VLZHBMLCSg1aYiUKPvYgoaj9KoMyRjrM5ixYJA8AdDmYKYYAAwAJLY+2k6uO10RJ3m0kvda32zxIChKpRV+uvYVN2PpDvs7npwelEA

Reply:
Ah, I see. The issue is that the temp variable `_a` isn't reserved in nested scopes.

Reply:
In src/compiler/transforms/classFields.ts, in the `createClassTempVar` function we have a call to `factory.createTempVariable` that isn't passing `true` for the `reservedInNestedScopes` parameter. I think that may be the culprit.

Reply:
```ts
const temp = factory.createTempVariable(requiresBlockScopedVar ? addBlockScopedVariable : hoistVariableDeclaration, !!isClassWithConstructorReference);
```

The `!!isClassWithConstructorReference` is too narrow. We may as well just always reserve the name.